site24x7/websitemonitor.go

```diff
@@ -207,199 +238,72 @@ func websiteMonitorCreateOrUpdate(method, url string, expectedResponseStatus int
 		i++
 	}
 
-	m := &WebsiteMonitor{
+	return &api.Monitor{
+		MonitorID:             d.Id(),
 		DisplayName:           d.Get("display_name").(string),
 		Type:                  "URL",
 		Website:               d.Get("website").(string),
 		CheckFrequency:        strconv.Itoa(d.Get("check_frequency").(int)),
 		HTTPMethod:            d.Get("http_method").(string),
 		AuthUser:              d.Get("auth_user").(string),
 		AuthPass:              d.Get("auth_pass").(string),
-		MatchingKeyword:       new(ValueAndSeverity),
-		UnmatchingKeyword:     new(ValueAndSeverity),
-		MatchRegex:            new(ValueAndSeverity),
+		MatchingKeyword:       new(api.ValueAndSeverity),
+		UnmatchingKeyword:     new(api.ValueAndSeverity),
+		MatchRegex:            new(api.ValueAndSeverity),
 		MatchCase:             d.Get("match_case").(bool),
 		UserAgent:             d.Get("user_agent").(string),
 		CustomHeaders:         customHeaders,
 		Timeout:               d.Get("timeout").(int),
 		LocationProfileID:     d.Get("location_profile_id").(string),
-		NotificationProfileID: d.Get("notification_profile_id").(string),
-		ThresholdProfileID:    d.Get("threshold_profile_id").(string),
+		NotificationProfileID: d.Get("NotificationProfileID").(string),
+		ThresholdProfileID:    d.Get("ThresholdProfileID").(string),
 		MonitorGroups:         monitorGroups,
 		UserGroupIDs:          userGroupIDs,
 		ActionIDs:             actionRefs,
 		UseNameServer:         d.Get("use_name_server").(bool),
 	}
-
-	if _, ok := d.GetOk("match_regex_value"); ok {
-		m.MatchRegex.Value = d.Get("match_regex_value").(string)
-		m.MatchRegex.Severity = Status(d.Get("match_regex_severity").(int))
-	} else {
-		m.MatchRegex = nil
-	}
-	if _, ok := d.GetOk("unmatching_keyword_value"); ok {
-		m.UnmatchingKeyword.Value = d.Get("unmatching_keyword_value").(string)
-		m.UnmatchingKeyword.Severity = Status(d.Get("unmatching_keyword_severity").(int))
-	} else {
-		m.UnmatchingKeyword = nil
-	}
-	if _, ok := d.GetOk("matching_keyword_value"); ok {
-		m.MatchingKeyword.Value = d.Get("matching_keyword_value").(string)
-		m.MatchingKeyword.Severity = Status(d.Get("matching_keyword_severity").(int))
-	} else {
-		m.MatchingKeyword = nil
-	}
-
-	if m.LocationProfileID == "" {
-		id, err := defaultLocationProfile(client)
-		if err != nil {
-			return err
-		}
-		m.LocationProfileID = id
-		d.Set("location_profile_id", id)
-	}
-	if m.NotificationProfileID == "" {
-		id, err := defaultNotificationProfile(client)
-		if err != nil {
-			return err
-		}
-		m.NotificationProfileID = id
-		d.Set("notification_profile_id", id)
-	}
-	if m.ThresholdProfileID == "" {
-		id, err := defaultThresholdProfile(client)
-		if err != nil {
-			return err
-		}
-		m.ThresholdProfileID = id
-		d.Set("threshold_profile_id", id)
-	}
-	if len(m.UserGroupIDs) == 0 {
-		id, err := defaultUserGroup(client)
-		if err != nil {
-			return err
-		}
-		m.UserGroupIDs = []string{id}
-		d.Set("user_group_ids", []string{id})
-	}
```

Comment:
The code you removed between line 244 and 295 is actually necessary and can be used almost as is. Basically you need to replace the calls to the `default*` funcs with the new `Default*` funcs we created in the previous PR. The original code can also be simplified a bit.